Mage.Tests/src/test/java/org/mage/test/cards/copy/CleverImpersonatorTest.java

```diff
@@ -180,10 +185,16 @@ Whenever enchanted land is tapped for mana, its controller adds two mana in any
         addCard(Zone.HAND, playerA, impersonator);
         addCard(Zone.BATTLEFIELD, playerA, "Island", 6);
         addCard(Zone.BATTLEFIELD, playerA, "Forest", 6);
+        addCard(Zone.BATTLEFIELD, playerA, "Maze of Ith",2,  true);
```

Comment:
This test will intermittently fail even when Maze of Ith is replaced with Mountain.

It fails because of a lack of color choices when a land enchanted by "Dawn's Reflection' is tapped for mana.

Something is going on, since it will fail due to those color choices even if the enchanted land is tapped, so that it could not be tapped for mana.

Reply:
1. That test generates a random results (you can see it in diff fail logs). Reason: cast uses random lands to pay, so it can be enchanted forest (with additional color dialogs) or can't be (without color dialog). Fix: uses `activateManaAbility` to pay correct colors. IntelliJ allows to run one test multiple times:
![shot_221004_105055](https://user-images.githubusercontent.com/8344157/193753912-fbe2bc8b-50fe-4b13-b318-88bfbebee8d3.png)

2. Correct code:
``` java
        activateManaAbility(1, PhaseStep.PRECOMBAT_MAIN, playerA, "{T}: Add {G}", 4);
        castSpell(1, PhaseStep.PRECOMBAT_MAIN, playerA, dReflection, "Forest"); // enchant a forest

        activateManaAbility(1, PhaseStep.POSTCOMBAT_MAIN, playerA, "{T}: Add {U}", 4);
        castSpell(1, PhaseStep.POSTCOMBAT_MAIN, playerA, impersonator);
        setChoice(playerA, true); // use Impersonator's effect
        setChoice(playerA, dReflection); // have Impersonator enter as copy of Dawn's Reflection
        setChoice(playerA, "Forest"); // enchant a forest

        setStrictChooseMode(true);
        setStopAt(1, PhaseStep.END_TURN);
        execute();
```

Reply:
I recently found out about running it multiple times, very useful.

My point was that it shouldn't be necessary to manually tap the mana. If I attach Dawn's Reflection to a tapped Mountain **tests should not be asking me for what color I pick** since that land is tapped, and so can't be tapped for mana.